code/createDeviceSettingsTable.m

```diff
@@ -0,0 +1,757 @@
+            HostUnixTime = currentSettings.RecordInfo.HostUnixTime;
+            Power_SettingsTable.action{entryNumber_Power} = sprintf('Stop Stream Power %d',streamStopCounter_Power);
+            Power_SettingsTable.recNum(entryNumber_Power) = streamStopCounter_Power;
+            Power_SettingsTable.time{entryNumber_Power} = HostUnixTime;
+            
+            % Fill in most recent power domain settings
+            Power_SettingsTable.powerBands{entryNumber_Power} = powerChannels;
+            for iChan = 1:4
+                TDsampleRates(iChan) = str2double(TDsettings(iChan).sampleRate(1:end-2));
+            end
+            TDsampleRates = unique(TDsampleRates);
+            currentTDsampleRate = TDsampleRates(~isnan(TDsampleRates));
+            Power_SettingsTable.TDsampleRates{entryNumber_Power} = currentTDsampleRate;
+            Power_SettingsTable.fftConfig(entryNumber_Power) = fftConfig;
+            
+            inStream_Power = 0;
+            streamStopCounter_Power = streamStopCounter_Power + 1;
+            entryNumber_Power = entryNumber_Power + 1;
+            
+        end
+        
+        % FFT
+        if inStream_FFT && ~currentSettings.StreamState.FftStreamEnabled
+            HostUnixTime = currentSettings.RecordInfo.HostUnixTime;
+            FFT_SettingsTable.action{entryNumber_FFT} = sprintf('Stop Stream FFT %d',streamStopCounter_FFT);
+            FFT_SettingsTable.recNum(entryNumber_FFT) = streamStopCounter_FFT;
+            FFT_SettingsTable.time{entryNumber_FFT} = HostUnixTime;
+            
+            % Fill in most recent FFT settings
+            FFT_SettingsTable.fftConfig(entryNumber_FFT) = fftConfig;
+            
+            % Get sample rate for each TD channel; all TD channels have
+            % same Fs (or is listed as NaN)
+            for iChan = 1:4
+                TDsampleRates(iChan) = str2double(TDsettings(iChan).sampleRate(1:end-2));
+            end
+            TDsampleRates = unique(TDsampleRates);
+            currentTDsampleRate = TDsampleRates(~isnan(TDsampleRates));
+            FFT_SettingsTable.TDsampleRates{entryNumber_FFT} = currentTDsampleRate;
+            
+            inStream_FFT = 0;
+            streamStopCounter_FFT = streamStopCounter_FFT + 1;
+            entryNumber_FFT = entryNumber_FFT + 1;
+        end
+    end
+    
+    % Option 2: Check if sense has been turned off
+    if isfield(currentSettings,'SenseState')
+        % TIME DOMAIN
+        if inStream_TD && isfield(currentSettings.SenseState,'state')
+            senseState = dec2bin(currentSettings.SenseState.state,4);
+            % Check starting/stopping of time domain streaming. See
+            % documentation enum Medtronic.NeuroStim.Olympus.DataTypes.Sensing.SenseStates : byte
+            % for more details about binary number coding
+            
+            if strcmp(senseState(4),'0') % Time domain streaming is off
```

Comment:
Want to make sure you are intentionally looking at the last bit. All even numbers up to 15 will have the same value at index 4.

Reply:
This goes for the other checks for the power and FFT table

Reply:
![image (4)](https://user-images.githubusercontent.com/3281849/96516693-779b9000-121c-11eb-84ab-e21450874691.png)

My understanding of this table is that 0000 indicates sense is off (for any/all streams). I have updated to ensure that all digits are 0 (so we don't mistakenly take something like 0100 -- power streaming on -- as streaming off). @roeegilron can you confirm this is the correct interpretation?

However, is `senseState = dec2bin(currentSettings.SenseState.state,4)` the correct way to parse the binary (e.g. what about cases when adaptive stim is on? Do we need to go to 8 bits?)

Changes made in 03ca94771d38ba12c52ba49ee2c17bea9121dee4

Reply:
Updated to 8 bit with 9ca2e4123fcb523466c7858059202091f284ce66